mlx/ops.cpp

```diff
@@ -1330,6 +1330,19 @@ array where(
       inputs);
 }
 
+array nan_to_num(
+    const array& a,
+    const array& nan,
+    const array& posinf,
+    const array& neginf,
+    StreamOrDevice s /* = {} */) {
+  auto out = where(isnan(a, s), nan, a, s);
```

Comment:
Probably should check the dtype of `a` here to see if it is inexact first. If it's not then just return `a` unmodified. Also worth adding a test case for that to make sure it works (e.g. test it when `a` has dtype `int32`.